File: Euclid/Messages/Incoming/SHOUT.cs
using Euclid.Game;
using Euclid.Network.Streams;
using Euclid.Util.Extensions;

namespace Euclid.Messages.Incoming
{
    class SHOUT : IMessageEvent
    {
        public void Handle(Player player, Request request)
        {
            if (!player.Authenticated)
            {
                return;
            }

            if (player.RoomEntity.Room == null)
            {
                return;
            }

            string message = request.Content.FilterInput();
            player.RoomEntity.Talk(ChatMessageType.SHOUT, message);
        }
    }
}

File: euclid/src/main/java/euclid/messages/incoming/SHOUT.java
package euclid.messages.incoming;

import euclid.game.Player;
import euclid.game.util.ChatMessageType;
import euclid.messages.IMessageEvent;
import euclid.network.streams.Request;
import euclid.util.extensions.StringExtensions;

public class SHOUT implements IMessageEvent {

    @Override
    public void handle(Player player, Request request) {
        if (!player.isAuthenticated()) {
            return;
        }

        if (player.getRoomEntity().getRoom() == null) {
            return;
        }

        String message = StringExtensions.filterInput(request.getContent());
        player.getRoomEntity().talk(ChatMessageType.SHOUT, message);
    }
}
